lib/ask_web/controllers/respondent_controller.ex

```diff
@@ -563,6 +564,38 @@ defmodule AskWeb.RespondentController do
     |> min(100.0)
   end
 
+  defp percentages(%{started_at: nil}), do: %{}
+
+  defp percentages(survey) do
+    %{
+      success_rate: cleanup_repetitive_percentages(Survey.success_rate_history(survey))
+    }
+  end
+
+  # Cleanup repetitive values, by skipping any repetitive value except for the
+  # first and last, because they don't contribute anything to the graph
+  # (straight line).
+  #
+  # For example take A, B, C, D consecutive dates, each at value 5.0 then we
+  # only keep A and D and skip B and C entirely.
+  #
+  # TODO: consider moving to RespondentView.
+  defp cleanup_repetitive_percentages(percentages, result \\ [])
```

Comment:
Please take the following comment totally optative. I'm pretty sure this code is fine as it was written. But I'm wondering while reading it: Do we need to program these operations? I'm just not sure. Maybe we do, but I see them too far from our domain, and pretty generic. Maybe we can take a similar logic from an external library, or push this logic outside this module, something like `cleanup_repetitive_values(list)` or something like that? What do you think?

Reply:
I didn't write this cleanup at first, and it was fine. But:

1. the methods for generating the cumulative percentages do such a cleanup, albeit a little differently (it does so while it sums the daily data, and only cares about straight lines);
2. the testsuite has _running_ surveys started in 2016, so we end up with over 2000 entries (:astonished:) in the JSON and it was impractical, so I did the cleanup :shrug:

I don't think we have other places where we cleanup historical chart data, so I think it's acceptable to have it here, especially since it's tied to how the data is structured (unless we decide to move it to the view).

Reply:
But I fully agree: it's not really at its place here. Maybe it ought to be moved to the View object.